Sources/BraintreeCore/BTHTTP.m

```diff
@@ -139,15 +149,56 @@ - (void)DELETE:(NSString *)aPath parameters:(NSDictionary *)parameters completio
 
 #pragma mark - Underlying HTTP
 
+- (void)httpRequestWithCaching:(NSString *)method path:(NSString *)aPath parameters:(NSDictionary *)parameters completion:(void(^)(BTJSON *body, NSHTTPURLResponse *response, NSError *error))completionBlock {
+    [self createRequest:method path:aPath parameters:parameters completion:^(NSURLRequest *request, NSError *error) {
+        if (error != nil) {
+            [self handleRequestCompletion:nil response:nil error:error completionBlock:completionBlock];
+            return;
+        }
+        
+        NSCachedURLResponse *cachedResponse = [[NSURLCache sharedURLCache] cachedResponseForRequest:request];
+        
+        if ([self.cacheDateValidator isCacheInvalid:cachedResponse]) {
+            [[NSURLCache sharedURLCache] removeAllCachedResponses];
+            cachedResponse = nil;
+        }
+
+        if (cachedResponse != nil) {
+            [self handleRequestCompletion:cachedResponse.data response:cachedResponse.response error:nil completionBlock:completionBlock];
+        } else {
+            NSURLSessionTask *task = [self.session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                if (data != nil && response != nil) {
+                    NSCachedURLResponse *cachedURLResponse = [[NSCachedURLResponse alloc]initWithResponse:response data:data];
+                    [[NSURLCache sharedURLCache] storeCachedResponse:cachedURLResponse forRequest:request];
```

Comment:
This could end up caching a failed response, like if it comes back as HTTP `5xx`.

Reply:
Good catch - [added a check here for this](https://github.com/braintree/braintree_ios/blob/4d649671887c2ea4da6486be37ed2a594b3925ba/Sources/BraintreeCore/BTHTTP.m#L342-L346)